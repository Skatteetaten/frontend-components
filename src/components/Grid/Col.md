**GridCol (Kolonne i rutenett): en hjelpekomponent til Grid**

```js noeditor
import { MessageBar } from '@skatteetaten/frontend-components/MessageBar';

<MessageBar>Det finnes ingen egne eksempler for GridCol.</MessageBar>;
```

```js noeditor beskrivelse
<div>
  <h3>Teknisk komponent for GridCol</h3>
  <p>
    GridCol er en hjelpekomponent til Grid som kan brukes til Ã¥ lage en knapp
    inni stripen.
  </p>
</div>
```
