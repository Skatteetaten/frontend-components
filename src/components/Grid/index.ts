export { Grid as default } from './Grid';
export * from './Grid';
