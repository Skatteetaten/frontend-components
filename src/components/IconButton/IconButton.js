import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { IconButton as FabricIconButton } from 'office-ui-fabric-react/lib-commonjs/Button';

import { getClassNames } from './IconButton.classNames';

/**
 * @visibleName IconButton (Ikonknapp)
 */
export default class IconButton extends React.PureComponent {
  static propTypes = {
    /** Gjør knappen til hovedknapp med fylt farge */
    buttonType: PropTypes.oneOf([
      'default',
      'small',
      'medium',
      'large',
      'xLarge'
    ]),
    /** Om knappen er inaktiv*/
    disabled: PropTypes.bool,
    /** Om sirkel skal vises eller ikke. Sirkel vil typisk benyttes når ikon fra material-design ikke har egen sirkel  */
    circle: PropTypes.bool,
    /** Ikon som skal vises foran teksten på knappen */
    icon: PropTypes.string,
    /** Beskrivelse av hva knappen gjør (f.eks. til skjermlesere) */
    title: PropTypes.string,
    /** Callback for klikk på knappen */
    onClick: PropTypes.func,
    className: PropTypes.string,
    /** Tekst alternativ til ikonet. Hvis bildet ikke har noen betydning vil attributten vise " " som default */
    alt: PropTypes.string
  };
  static defaultProps = {
    buttonType: 'default',
    circle: false,
    icon: undefined,
    title: undefined,
    disabled: undefined,
    onClick: undefined,
    alt: ' '
  };

  render() {
    const {
      buttonType,
      icon,
      title,
      disabled,
      onClick,
      className,
      alt,
      ...props
    } = this.props;
    return (
      <FabricIconButton
        {...props}
        className={classnames(getClassNames(this.props), className)}
        buttonType={buttonType}
        title={title}
        aria-label={title}
        role="button"
        iconProps={{ iconName: icon }}
        disabled={disabled}
        onClick={onClick}
        alt={alt}
      />
    );
  }
}
