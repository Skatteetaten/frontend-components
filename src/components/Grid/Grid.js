import React from 'react';
import PropTypes from 'prop-types';
import { getClassNames } from './Grid.classNames';

class Row extends React.Component {
  static displayName = 'GridRow';

  static propTypes = {
    rowSpacing: PropTypes.string,
    rowInset: PropTypes.bool,
    centered: PropTypes.bool
  };

  static defaultProps = {
    rowSpacing: '8px'
  };

  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).row} ${className}` },
      children
    );
  }
}

class Col extends React.Component {
  static displayName = 'GridCol';
  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).col} ${className}` },
      children
    );
  }
}

Col.propTypes = {
  noSpacing: PropTypes.bool
};

/**
 * @visibleName Grid (Rutenett)
 */
export class Grid extends React.Component {
  static Col = Col;
  static Row = Row;

  static SPACE_NONE = '0px';
  static SPACE_SMALL = '8px';
  static SPACE_MEDIUM = '16px';
  static SPACE_LARGE = '24px';

  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).grid} ${className}` },
      children
    );
  }
}

export default Grid;
