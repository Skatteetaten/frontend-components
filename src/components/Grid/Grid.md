** Grid er en komponent som brukes til å plassere komponenter og annet innhold på siden, og hjelper til med å justere plasseringen for ulike skjermbredder. **

```js noeditor
DemoBlock = ({ children }) => (
  <div
    style={{
      border: '1px solid black',
      backgroundColor: '#ddd',
      textAlign: 'center',
      marginBottom: '0',
      padding: '8px 0'
    }}
  >
    {children}
  </div>
);
```

```js
import Grid from '@skatteetaten/frontend-components/Grid';

<Grid>
  <Grid.Row>
    <Grid.Col lg={4}>
      <DemoBlock>4</DemoBlock>
    </Grid.Col>
    <Grid.Col lg={4}>
      <DemoBlock>4</DemoBlock>
    </Grid.Col>
    <Grid.Col lg={4}>
      <DemoBlock>4</DemoBlock>
    </Grid.Col>
  </Grid.Row>
  <Grid.Row>
    <Grid.Col lg={8}>
      <DemoBlock>8</DemoBlock>
    </Grid.Col>
    <Grid.Col lg={4}>
      <DemoBlock>4</DemoBlock>
    </Grid.Col>
  </Grid.Row>
  <Grid.Row>
    <Grid.Col lg={6}>
      <DemoBlock>6</DemoBlock>
    </Grid.Col>
    <Grid.Col lg={6}>
      <DemoBlock>6</DemoBlock>
    </Grid.Col>
  </Grid.Row>
  <Grid.Row rowSpacing={Grid.SPACE_LARGE}>
    <Grid.Col lg={12}>
      <DemoBlock>12</DemoBlock>
    </Grid.Col>
  </Grid.Row>
</Grid>;
```

```js noeditor
import Accordion from '@skatteetaten/frontend-components/Accordion';
import AccordionItem from '@skatteetaten/frontend-components/Accordion/AccordionItem';
<Accordion>
  <AccordionItem
    isOpen
    toggleContent
    toggleButtonText={'Bruk'}
    stepId={'step-1-1'}
  >
    <p>
      Et grid/rutenett hjelper til med å plassere komponenter på skjermen. Det
      er først og fremst nyttig i utvikling, men kan være lurt for designere å
      tenke på hvordan komponenter enkelt plasseres
    </p>
    <p>
      Et grid består av rader og kolonner, hvor en rad kan maksimalt ha 12
      kolonner, men man trenger ikke bruke alle 12
    </p>
    <p>
      Griden kan settes opp slik at antall kolonner som vises er avhengig av
      skjermstørrelsen, og på den måten kan realisere responsivt design. Vi har
      følgende brekkpunkter:
    </p>
    <ul>
      <li>Small (sm): maks 479px</li>
      <li>Medium (md): 480px-639px</li>
      <li>Large (lg): 640px-1023px</li>
      <li>X-large (xl): 1024px-1365px</li>
      <li>Xx-large (xxl): 1366px-1919px</li>
      <li>Xxx-large (xxxl): min 1920px</li>
    </ul>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Universell utforming'}
    stepId={'step-1-2'}
  >
    <p>Dette seksjonen er foreløpig tom.</p>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Tekniske detaljer'}
    stepId={'step-1-3'}
  >
    <p>
      Rader har standard avstand på 8px. Men kan endre denne per rad ved å bruke
      rowSpacing: `Grid.SPACE_NONE, Grid.SPACE_SMALL, Grid.SPACE_MEDIUM,
      Grid.SPACE_LARGE`
    </p>
  </AccordionItem>
</Accordion>;
```
