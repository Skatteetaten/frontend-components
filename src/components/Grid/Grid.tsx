import * as React from 'react';
import { getClassNames } from './Grid.classNames';

interface RowProps {
  rowSpacing?: string;
  rowInset?: boolean;
  centered?: boolean;
  className?: string;
  tag?: string;
}

class Row extends React.Component<RowProps, {}> {
  static displayName = 'GridRow';

  static defaultProps = {
    rowSpacing: '8px'
  };

  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).row} ${className}` },
      children
    );
  }
}

interface ColProps {
  noSpacing?: boolean;
  className?: string;
  tag?: string;
  sm?: number;
  md?: number;
  lg?: number;
  xl?: number;
  xxl?: number;
  xxxl?: number;
  smPush?: number;
  mdPush?: number;
  lgPush?: number;
  xlPush?: number;
  xxlPush?: number;
  xxxlPush?: number;
  smPull?: number;
  mdPull?: number;
  lgPull?: number;
  xlPull?: number;
  xxlPull?: number;
  xxxlPull?: number;
}
class Col extends React.Component<ColProps, {}> {
  static displayName = 'GridCol';
  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).col} ${className}` },
      children
    );
  }
}

interface GridProps {
  className?: string;
  tag?: string;
}
/**
 * @visibleName Grid (Rutenett)
 */
export class Grid extends React.Component<GridProps, {}> {
  static Col = Col;
  static Row = Row;
  static SPACE_NONE = '0px';
  static SPACE_SMALL = '8px';
  static SPACE_MEDIUM = '16px';
  static SPACE_LARGE = '24px';

  render() {
    const { children, className = '', tag = 'div' } = this.props;
    return React.createElement(
      tag,
      { className: `${getClassNames(this.props).grid} ${className}` },
      children
    );
  }
}

export default Grid;
