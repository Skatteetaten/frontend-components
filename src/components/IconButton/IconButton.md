** Ikonknapper brukes til mindre fremtredende funksjoner, når man har lite plass eller når funksjonen er innlysende og man ikke trenger å kommunisere den tekstlig **

```js
import IconButton from '@skatteetaten/frontend-components/IconButton';

<div className="ExampleSpacing8">
  <IconButton
    alt={'Sky med sirkel rundt'}
    title="Skriv ut"
    circle
    type="large"
    icon="Print"
  />
  <IconButton
    alt={'Sky med sirkel rundt'}
    title="Skriv ut"
    type="large"
    icon="Print"
  />
  <br />
  <IconButton
    alt={'Sky med sirkel rundt'}
    title="Skriv ut"
    circle
    type="default"
    icon="Print"
  />
  <IconButton
    alt={'Sky med sirkel rundt'}
    title="Skriv ut"
    type="default"
    icon="Print"
  />
</div>;
```

```js noeditor
import Accordion from '@skatteetaten/frontend-components/Accordion';
import AccordionItem from '@skatteetaten/frontend-components/Accordion/AccordionItem';

<Accordion>
  <AccordionItem
    isOpen
    toggleContent
    toggleButtonText={'Bruk'}
    stepId={'step-1-1'}
  >
    <p>
      Knapper bør i størst mulig grad ha den beskrivende teksten synlig, men
      ikonknapper kan brukes i situasjoner der man ikke har anleding til å vise
      den. Det skal alltid være en tekst som kommer ved mouseover som forklarer
      hva ikonet gjør.
    </p>
    <p>Ikonknapper bør brukes med varsomhet i publikumsløsninger.</p>
    <p>
      Ikonknapper kan være små eller store, med eller uten sirkel. Sirkelen
      benyttes for å fremheve ikonknappen.
    </p>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Universell utforming'}
    stepId={'step-1-2'}
  >
    <ul>
      <li>Tittel på knappen bør beskrive dens funksjon.</li>
      <li>
        Husk å bruke gi ikonknappen en label slik at skjermlesere kan lese opp
        funksjonen
      </li>
    </ul>
  </AccordionItem>
  <AccordionItem
    toggleContent
    toggleButtonText={'Tekniske detaljer'}
    stepId={'step-1-3'}
  >
    <p>
      <a
        href="https://developer.microsoft.com/en-us/fabric#/components/button#Implementation"
        target="_blank"
      >
        Se flere tilgjengelige props i Fabric dokumentasjonen
      </a>
    </p>
  </AccordionItem>
</Accordion>;
```
