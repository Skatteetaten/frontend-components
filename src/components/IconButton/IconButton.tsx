import classnames from 'classnames';
import * as React from 'react';

import {
  IconButton as FabricIconButton,
  IButtonProps
} from 'office-ui-fabric-react/lib-commonjs/Button';

import { getClassNames } from './IconButton.classNames';

interface IconButtonProps extends IButtonProps {
  /** Om sirkel skal vises eller ikke. Sirkel vil typisk benyttes når ikon fra material-design ikke har egen sirkel  */
  circle?: boolean;
  /** Ikon som skal vises foran teksten på knappen */
  icon?: string;
  /** Gjør knappen til hovedknapp med fylt farge */
  type?: 'default' | 'small' | 'medium' | 'large' | 'xLarge';
}
/**
 * @visibleName IconButton (Ikonknapp)
 */
export default class IconButton extends React.PureComponent<
  IconButtonProps,
  {}
> {
  static defaultProps = {
    alt: ' ',
    circle: false,
    disabled: undefined,
    icon: undefined,
    onClick: undefined,
    title: undefined,
    type: 'default'
  };

  render() {
    const { icon, className, ...props } = this.props;
    return (
      <FabricIconButton
        {...props}
        className={classnames(getClassNames(this.props), className)}
        role="button"
        iconProps={{ iconName: icon }}
      />
    );
  }
}
