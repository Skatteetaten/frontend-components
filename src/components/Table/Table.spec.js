import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Table } from '.';
import { IconButton } from '../IconButton';
import { Grid } from '../Grid';

const data = [
  {
    Måned: 'Januar',
    Beløp: 100,
    Dekningsgrad: '100%',
    Avkastning: '1000',
    ikkeTegnetVerdi: '2000',
  },
  {
    Måned: 'Februar',
    Beløp: 100,
    Dekningsgrad: '50%',
    Avkastning: '500',
  },
  {
    Måned: 'Mars',
    Beløp: 100,
    Dekningsgrad: '20%',
    Avkastning: '2000',
  },
  {
    Måned: 'April',
    Beløp: 100,
    Dekningsgrad: '30%',
    Avkastning: '1055',
  },
];

const columns = [
  {
    name: 'Måned',
    fieldName: 'month',
  },
  {
    name: 'Beløp',
    fieldName: 'amount',
    alignment: 'right',
  },
  {
    name: 'Dekningsgrad',
    fieldName: 'coverage',
    alignment: 'right',
  },
  {
    name: 'Avkastning',
    fieldName: 'revenue',
    alignment: 'right',
  },
];

const content = (close) => (
  <div>
    <p>
      <strong>{data.Måned}</strong>
    </p>

    <Grid>
      <Grid.Row>
        <Grid.Col lg={12}>
          <p>Her kommer redigerbart innhold</p>
        </Grid.Col>
        <Grid.Col lg={3}>
          <div>
            <IconButton
              onClick={close}
              title="Angre"
              id="closeEditMode"
              circle
              icon="Cancel"
            />{' '}
          </div>
        </Grid.Col>
      </Grid.Row>
    </Grid>
  </div>
);

function oppsettMount(props) {
  return mount(<Table {...props} />);
}

function mountMedEditerbartInnholdAapen() {
  const wrapper = oppsettMount({
    data,
    columns,
    editableRows: true,
    editableContent: content,
  });

  wrapper.find('TableRow').first().find('IconButton').first().simulate('click');

  return wrapper;
}

describe('Table komponent', () => {
  it('matcher snapshot', () => {
    const wrapper = oppsettMount({ data, columns });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('rendrer Table med riktig props', () => {
    const wrapper = oppsettMount({
      data,
      columns,
      editableRows: true,
      id: 'tableid',
      customClassNames: { wrapper: 'tableClass' },
      editableContent: 'Editerbart innhold',
    });

    const tableWrapper = wrapper.find('Table > div');
    const tableRow = wrapper.find('TableRow').first();

    expect(tableWrapper.prop('id')).toEqual('tableid');
    expect(tableWrapper.prop('className')).toContain('tableClass');
    expect(tableRow.prop('editableRows')).toEqual(true);
    expect(tableRow.prop('editableContent')).toEqual('Editerbart innhold');
  });

  it('rendrer Table med riktig antall rader ', () => {
    const wrapper = oppsettMount({ data, columns });

    expect(wrapper.find('TableRow').length).toEqual(4);
  });

  it('rendrer Table med editerbare rader ', () => {
    const wrapper = oppsettMount({ data, columns, editableRows: true });
    const tableRow = wrapper.find('TableRow');

    expect(wrapper.find('thead').exists('button')).toEqual(false);
    expect(tableRow.at(1).exists('button')).toEqual(true);
    expect(tableRow.at(2).exists('button')).toEqual(true);
  });

  it('viser editerbart innhold når editeringsknapp for en tabellrad klikkes ', () => {
    const wrapper = oppsettMount({
      data,
      columns,
      editableRows: true,
      editableContent: content,
    });

    const editButton = wrapper
      .find('TableRow')
      .first()
      .find('IconButton')
      .first();

    expect(wrapper.html()).not.toContain('Her kommer redigerbart innhold');
    editButton.simulate('click');
    expect(wrapper.html()).toContain('Her kommer redigerbart innhold');
  });

  it('viser editerbart innhold når en tabellrad klikkes med openEditableOnRowClick', () => {
    const wrapper = oppsettMount({
      data,
      columns,
      editableRows: true,
      editableContent: content,
      openEditableOnRowClick: true,
    });

    const klikkbarTabellCelle = wrapper
      .find('TableRow')
      .first()
      .find('[data-testid="openEditableOnRowClick-button"]')
      .first();

    expect(wrapper.html()).not.toContain('Her kommer redigerbart innhold');
    klikkbarTabellCelle.simulate('click');
    expect(wrapper.find('TableRow').first().html()).toContain(
      'Her kommer redigerbart innhold'
    );
  });

  it('setter korrekt colspan på editerbart innhold sin kolonne', () => {
    const wrapper = mountMedEditerbartInnholdAapen();

    const editableContentColspan = wrapper
      .find('[data-testid="editable-content"]')
      .getDOMNode()
      .getAttribute('colspan');

    expect(editableContentColspan).toBe('5');
  });

  it('når en tabellrad er i editeringsmodus skal det ikke være mulig å editere øvrige rader ', () => {
    const wrapper = mountMedEditerbartInnholdAapen();

    wrapper.find('.editButton').forEach((node) => {
      expect(node.prop('disabled')).toEqual(true);
    });
  });
  it('rendrer Table med ekspanderbare rader ', () => {
    const wrapper = oppsettMount({ data, columns, expandableRows: true });
    const tableRow = wrapper.find('TableRow');
    expect(wrapper.find('thead').exists('button')).toEqual(false);
    expect(tableRow.at(1).exists('button')).toEqual(true);
    expect(tableRow.at(2).exists('button')).toEqual(true);
  });
  it('viser ekspanderbart innhold når ekspanderingsknapp for en tabellrad klikkes', () => {
    const mockContent = (mockdata, close, rowIndex) => (
      <div className={'mockDiv'}>
        Ekspanderbart innhold for {mockdata.Måned}
      </div>
    );
    const wrapper = oppsettMount({
      data,
      columns,
      expandableRows: true,
      expandableContent: mockContent,
    });
    const tableRow = wrapper.find('TableRow');
    expect(wrapper.exists('.mockDiv')).toEqual(false);
    tableRow.at(0).find('IconButton').first().simulate('click');
    expect(wrapper.exists('.mockDiv')).toEqual(true);
    expect(wrapper.find('.mockDiv').text()).toEqual(
      'Ekspanderbart innhold for Januar'
    );
  });

  it('håndterer openEditableRowIndex fra props', () => {
    const wrapper = mount(
      <Table
        data={data}
        columns={columns}
        editableContent={() => <div id="edit">{content.Måned}</div>}
        editableRows={true}
        openEditableRowIndex={0}
      />
    );

    let tableRows = wrapper.find('TableRow');
    expect(tableRows.at(0).exists('#edit')).toEqual(true);
  });

  it('tegner Table med en rad og en underrad', () => {
    const dataMedUnderlinjer = [
      {
        Måned: 'Februar',
        Beløp: 100,
        Dekningsgrad: '50%',
        Avkastning: '500',
        children: [
          {
            Beløp: 1000,
            Dekningsgrad: '50%',
            Avkastning: '5000',
          },
        ],
      },
    ];
    const wrapper = oppsettMount({
      data: dataMedUnderlinjer,
      columns,
      editableRows: true,
      id: 'tableid',
      className: 'tableClass',
      editableContent: 'Editerbart innhold',
    });
    expect(wrapper.find('TableRow').length).toEqual(1);

    const tableRow = wrapper.find('TableRow');
    expect(tableRow.find('tr').length).toEqual(2);
  });
  it('skal vise sum og opprette tomme <td> dersom kolonneheaderen er tom', () => {
    const wrapper = oppsettMount({
      data,
      columns: [
        ...columns,
        {
          name: '',
          fieldName: 'empty1',
        },
        {
          name: '',
          fieldName: 'empty2',
        },
      ],
      id: 'tableid',
      className: 'tableClass',
      sum: { text: 'Sum:', colspan: 3, total: '4555' },
    });
    expect(wrapper.find('SumRow').text()).toEqual('Sum:4555');
    expect(wrapper.find('SumRow').find('td').length).toEqual(3);
  });
});
