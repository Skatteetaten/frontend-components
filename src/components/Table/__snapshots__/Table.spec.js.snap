// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table komponent matcher snapshot 1`] = `
<Table
  columns={
    Array [
      Object {
        "fieldName": "month",
        "name": "Måned",
      },
      Object {
        "alignment": "right",
        "fieldName": "amount",
        "name": "Beløp",
      },
      Object {
        "alignment": "right",
        "fieldName": "coverage",
        "name": "Dekningsgrad",
      },
      Object {
        "alignment": "right",
        "fieldName": "revenue",
        "name": "Avkastning",
      },
    ]
  }
  data={
    Array [
      Object {
        "Avkastning": "1000",
        "Beløp": 100,
        "Dekningsgrad": "100%",
        "Måned": "Januar",
      },
      Object {
        "Avkastning": "500",
        "Beløp": 100,
        "Dekningsgrad": "50%",
        "Måned": "Februar",
      },
      Object {
        "Avkastning": "2000",
        "Beløp": 100,
        "Dekningsgrad": "20%",
        "Måned": "Mars",
      },
      Object {
        "Avkastning": "1055",
        "Beløp": 100,
        "Dekningsgrad": "30%",
        "Måned": "April",
      },
    ]
  }
>
  <div
    className="SkeTable-40"
  >
    <table>
      <thead>
        <tr>
          <th
            className=""
            key="month"
          >
            Måned
          </th>
          <th
            className=""
            key="amount"
          >
            Beløp
          </th>
          <th
            className=""
            key="coverage"
          >
            Dekningsgrad
          </th>
          <th
            className=""
            key="revenue"
          >
            Avkastning
          </th>
        </tr>
      </thead>
      <tbody>
        <TableRow
          columns={
            Array [
              Object {
                "fieldName": "month",
                "name": "Måned",
              },
              Object {
                "alignment": "right",
                "fieldName": "amount",
                "name": "Beløp",
              },
              Object {
                "alignment": "right",
                "fieldName": "coverage",
                "name": "Dekningsgrad",
              },
              Object {
                "alignment": "right",
                "fieldName": "revenue",
                "name": "Avkastning",
              },
            ]
          }
          data={
            Object {
              "Avkastning": "1000",
              "Beløp": 100,
              "Dekningsgrad": "100%",
              "Måned": "Januar",
            }
          }
          editModeActive={false}
          key="0"
          rowIndex={0}
          setEditMode={[Function]}
          tableHasScroll={false}
        >
          <tr
            key="0"
          >
            <td
              className=""
              key="0_0"
            />
            <td
              className="right"
              key="0_1"
            />
            <td
              className="right"
              key="0_2"
            />
            <td
              className="right"
              key="0_3"
            />
          </tr>
        </TableRow>
        <TableRow
          columns={
            Array [
              Object {
                "fieldName": "month",
                "name": "Måned",
              },
              Object {
                "alignment": "right",
                "fieldName": "amount",
                "name": "Beløp",
              },
              Object {
                "alignment": "right",
                "fieldName": "coverage",
                "name": "Dekningsgrad",
              },
              Object {
                "alignment": "right",
                "fieldName": "revenue",
                "name": "Avkastning",
              },
            ]
          }
          data={
            Object {
              "Avkastning": "500",
              "Beløp": 100,
              "Dekningsgrad": "50%",
              "Måned": "Februar",
            }
          }
          editModeActive={false}
          key="1"
          rowIndex={1}
          setEditMode={[Function]}
          tableHasScroll={false}
        >
          <tr
            key="1"
          >
            <td
              className=""
              key="1_0"
            />
            <td
              className="right"
              key="1_1"
            />
            <td
              className="right"
              key="1_2"
            />
            <td
              className="right"
              key="1_3"
            />
          </tr>
        </TableRow>
        <TableRow
          columns={
            Array [
              Object {
                "fieldName": "month",
                "name": "Måned",
              },
              Object {
                "alignment": "right",
                "fieldName": "amount",
                "name": "Beløp",
              },
              Object {
                "alignment": "right",
                "fieldName": "coverage",
                "name": "Dekningsgrad",
              },
              Object {
                "alignment": "right",
                "fieldName": "revenue",
                "name": "Avkastning",
              },
            ]
          }
          data={
            Object {
              "Avkastning": "2000",
              "Beløp": 100,
              "Dekningsgrad": "20%",
              "Måned": "Mars",
            }
          }
          editModeActive={false}
          key="2"
          rowIndex={2}
          setEditMode={[Function]}
          tableHasScroll={false}
        >
          <tr
            key="2"
          >
            <td
              className=""
              key="2_0"
            />
            <td
              className="right"
              key="2_1"
            />
            <td
              className="right"
              key="2_2"
            />
            <td
              className="right"
              key="2_3"
            />
          </tr>
        </TableRow>
        <TableRow
          columns={
            Array [
              Object {
                "fieldName": "month",
                "name": "Måned",
              },
              Object {
                "alignment": "right",
                "fieldName": "amount",
                "name": "Beløp",
              },
              Object {
                "alignment": "right",
                "fieldName": "coverage",
                "name": "Dekningsgrad",
              },
              Object {
                "alignment": "right",
                "fieldName": "revenue",
                "name": "Avkastning",
              },
            ]
          }
          data={
            Object {
              "Avkastning": "1055",
              "Beløp": 100,
              "Dekningsgrad": "30%",
              "Måned": "April",
            }
          }
          editModeActive={false}
          key="3"
          rowIndex={3}
          setEditMode={[Function]}
          tableHasScroll={false}
        >
          <tr
            key="3"
          >
            <td
              className=""
              key="3_0"
            />
            <td
              className="right"
              key="3_1"
            />
            <td
              className="right"
              key="3_2"
            />
            <td
              className="right"
              key="3_3"
            />
          </tr>
        </TableRow>
      </tbody>
    </table>
  </div>
</Table>
`;
